pkg/operator/cassandra/controller/service.go

```diff
@@ -0,0 +1,135 @@
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/controller/endpoint"
+	"strings"
+)
+
+const (
+	unusedPort = 64551
+)
+
+// SyncClusterHeadlessService checks if a Headless Service exists
+// for the given Cluster, in order for the StatefulSets to utilize it.
+// If it doesn't exists, then create it.
+func (cc *ClusterController) SyncClusterHeadlessService(c *cassandrav1alpha1.Cluster) error {
+	clusterHeadlessService := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            util.HeadlessServiceNameForCluster(c),
+			Namespace:       c.Namespace,
+			Labels:          util.ClusterLabels(c.Name),
+			OwnerReferences: []metav1.OwnerReference{util.NewControllerRef(c)},
+		},
+		Spec: corev1.ServiceSpec{
+			ClusterIP: corev1.ClusterIPNone,
+			Type:      corev1.ServiceTypeClusterIP,
+			Selector:  util.ClusterLabels(c.Name),
+			// Necessary to specify a Port to work correctly
+			// https://github.com/kubernetes/kubernetes/issues/32796
+			Ports: []corev1.ServicePort{{Port: 9042}},
+		},
+	}
+
+	return cc.syncService(clusterHeadlessService, c)
+}
+
+// SyncMemberServices checks, for every Pod of the Cluster that
+// has been created, if a corresponding ClusterIP Service exists,
+// which will serve as a static ip.
+// If it doesn't exist, it creates it.
+func (cc *ClusterController) SyncMemberServices(c *cassandrav1alpha1.Cluster) error {
+
+	pods, err := util.GetPodsForCluster(c, cc.podLister)
+	if err != nil {
+		return err
+	}
+
+	// For every Pod of the cluster that exists, check that a
+	// a corresponding ClusterIP Service exists, and if it doesn't,
+	// create it.
+	for _, pod := range pods {
+		if err := cc.syncService(memberServiceForPod(pod, c), c); err != nil {
+			logger.Errorf("Error syncing member service for '%s'", pod.Name)
+			return err
+		}
+	}
+	return nil
+}
+
+// syncService checks if the given Service exists and creates it if it doesn't
+// it creates it
+func (cc *ClusterController) syncService(s *corev1.Service, c *cassandrav1alpha1.Cluster) error {
+
+	existingService, err := cc.serviceLister.Services(s.Namespace).Get(s.Name)
+	// If we get an error but without the IsNotFound error raised
+	// then something is wrong with the network, so requeue.
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	// If the service already exists, check that it's
+	// controlled by the given Cluster
+	if err == nil {
+		return util.VerifyOwner(existingService, c)
+	}
+
+	// At this point, the Service doesn't exist, so we are free to create it
+	_, err = cc.kubeClient.CoreV1().Services(s.Namespace).Create(s)
+	return err
+
+}
+
+func memberServiceForPod(pod *corev1.Pod, cluster *cassandrav1alpha1.Cluster) *corev1.Service {
+
+	// If Member is seed, add the appropriate label
+	labels := util.ClusterLabels(cluster.Name)
+	if strings.HasSuffix(pod.Name, "-0") || strings.HasSuffix(pod.Name, "-1") {
```

Comment:
is this saying that the first 2 pods in the statefulset are "seed" members? is that always true, or is the number of seed members configurable?

Reply:
Exactly ! I will document that better. 
For the time being, this isn't configurable (and I'm not too sure it even needs to be).